filter/metrics_parser_test.go

```diff
@@ -74,6 +75,26 @@ func TestParseMetric(t *testing.T) {
 			So(parsedMetric.Value, ShouldEqual, validMetric.value)
 			So(parsedMetric.Timestamp, ShouldEqual, validMetric.timestamp)
 		}
+
+		Convey("Check metrics with magic '-1' timestamp", func() {
+			timeStart := time.Now().Unix()
+			magicTimestampMetrics := []ValidMetricCase{
+				{"One.two.three 123 -1", "One.two.three", "One.two.three", map[string]string{}, 123, timeStart},
+				{"One.two.three 1.23e2 -1", "One.two.three", "One.two.three", map[string]string{}, 123, timeStart},
+				{"One.two.three -123 -1", "One.two.three", "One.two.three", map[string]string{}, -123, timeStart},
+			}
+
+			for _, magicMetric := range magicTimestampMetrics {
+				parsedMetric, err := ParseMetric([]byte(magicMetric.input))
```

Comment:
I would recommend you to wrap each test case in different `Convey` call with valid name to be able to differentiate test cases in tests run report.